util/network-devp2p/src/disk.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+// This file is part of OpenEthereum.
+
+// OpenEthereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// OpenEthereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Open Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use log::*;
+use parity_crypto::publickey::Secret;
+use parity_path::restrict_permissions_owner;
+use std::fmt::Debug;
+use std::fs;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+
+pub trait DiskEntity: FromStr {
```

Comment:
If this trait is intended to be an abstraction over `a file` it would be better to move somewhere else because it's not really related to `devp2p networking` but for now it is fine.